pydantic/json_schema.py

```diff
@@ -969,6 +970,11 @@ def dict_schema(self, schema: core_schema.DictSchema) -> JsonSchemaValue:
             else:
                 json_schema['patternProperties'] = {keys_pattern: values_schema}
 
+        if ks is not None:
+            keys_type = ks.get('type', None)
+            if keys_type == 'enum':
+                keys_members = ks.get('members', [])
+                json_schema['propertyNames'] = {'enum': keys_members}
```

Comment:
I think we can add support for any key schema out of the box

```suggestion
        # Add `keys_schema.pop('title', None)` next to `values_schema.pop('title', None)` above and update comment
        if keys_schema.get('type') == 'string' and len(keys_schema) > 1:  # len check means there are additional constraints to the key schema
            json_schema['propertyNames'] = keys_schema
```

?

Reply:
Thanks for the review and the suggestion :) Just for better understanding from my side:
keys_schema could be a ref only, which means that I would need to resolve that before I can do any checks on that, right?
Would it then be best to add the ref to propertyNames schema, or the resolved one? I'll go with the resolved for now but am open for your inputs :)

Reply:
While doing it, this will also add a propertyNames schema to an existing test case. This should be expected behaviour, as python's Path in json schema of pydantic has the ' format' 'path' schema.

I just have to add this to the existing test case, and I'm not sure how such changes to existing behaviour is handled in pydantic tbh :)

Reply:
> Would it then be best to add the ref to propertyNames schema, or the resolved one?

Looked closer at this - I think we can avoid the ref resolution. Can change if it comes up for a specific use case.